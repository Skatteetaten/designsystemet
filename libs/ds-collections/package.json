{
  "name": "@skatteetaten/ds-collections",
  "groupId": "no.skatteetaten.aurora",
  "version": "1.4.0",
  "license": "Apache-2.0",
  "homepage": "https://www.skatteetaten.no/stilogtone/designsystemet/",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Skatteetaten/designsystemet.git"
  },
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "i18next": "^23",
    "react": "^18",
    "react-i18next": "^14",
    "@skatteetaten/ds-core-utils": "1.3.0",
    "@skatteetaten/ds-icons": "1.3.0",
    "@skatteetaten/ds-buttons": "1.3.0",
    "@skatteetaten/ds-content": "1.3.0",
    "@skatteetaten/ds-typography": "1.3.0",
    "@skatteetaten/ds-progress": "1.3.0"
  },
  "scripts": {}
}
