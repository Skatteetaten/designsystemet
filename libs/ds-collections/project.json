{
  "name": "ds-collections",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-collections/src",
  "projectType": "library",
  "tags": [],
  "implicitDependencies": ["!ds-dev-config"],
  "targets": {
    "lint": {
      "executor": "@nx/eslint:lint",
      "outputs": ["{options.outputFile}"]
    },
    "build": {
      "executor": "@nx/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-collections",
        "tsConfig": "libs/ds-collections/tsconfig.lib.json",
        "project": "libs/ds-collections/package.json",
        "entryFile": "libs/ds-collections/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": ["libs/ds-collections/rollup.config.js"],
        "compiler": "swc",
        "assets": [
          {
            "glob": "libs/ds-collections/README.md",
            "input": ".",
            "output": "."
          },
          {
            "glob": "LICENSE",
            "input": ".",
            "output": "."
          }
        ],
        "babelUpwardRootMode": true
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/ds-collections/jest.config.ts"
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-collections/**/*.css",
          "libs/ds-collections/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "@skatteetaten/ds-dev-config:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      },
      "dependsOn": [
        {
          "projects": "ds-core-designtokens",
          "target": "build"
        }
      ]
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-collections"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish dist/libs/ds-collections {args.switches}",
        "cwd": ""
      }
    }
  }
}
