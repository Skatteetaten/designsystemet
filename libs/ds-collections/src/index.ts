export * from './Accordion/Accordion';
export * from './Accordion/Accordion.types';
export * from './AccordionItem/AccordionItem.types';
export * from './Chips/Chips';
export * from './Chips/Chips.types';
export * from './ChipsToggle/ChipsToggle';
export * from './ChipsToggle/ChipsToggle.types';
export * from './ChipsRemovable/ChipsRemovable';
export * from './ChipsRemovable/ChipsRemovable.types';
export * from './OpenClose/OpenClose';
export * from './OpenClose/OpenClose.types';
export * from './StepList/StepList';
export * from './StepList/StepList.types';
export * from './StepListStep/StepListStep.types';
export * from './Tabs/Tabs';
export * from './Tabs/Tabs.types';
