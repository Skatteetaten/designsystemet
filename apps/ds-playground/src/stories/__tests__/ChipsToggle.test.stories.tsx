import { Chips, ChipsToggleProps } from '@skatteetaten/ds-collections';
import { Meta, StoryObj } from '@storybook/react';
import { expect, within } from '@storybook/test';

const elementId = 'htmlId';

const meta = {
  component: Chips.Toggle,
  title: 'Tester/Chips/Toggle',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    size: { table: { disable: true } },
    isSelected: { table: { disable: true } },
    showCheckmark: { table: { disable: true } },
    children: { table: { disable: true } },
    // HTML
    disabled: { table: { disable: true } },
    // Events
    onClick: { table: { disable: true } },
  },
} satisfies Meta<typeof Chips.Toggle>;
export default meta;
type Story = StoryObj<typeof meta>;

const defaultArgs: ChipsToggleProps = {
  children: 'Trondheim',
};

export const WithRef = {
  name: 'With Ref (FA1)',
  args: {
    ...defaultArgs,
    ref: (instance: HTMLButtonElement | null): void => {
      if (instance) {
        instance.id = 'dummyIdForwardedFromRef';
      }
    },
  },
  argTypes: {
    ref: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const container = canvas.getByRole('button');
    await expect(container).toHaveAttribute('id', 'dummyIdForwardedFromRef');
  },
} satisfies Story;

export const WithAttributes = {
  name: 'With Attributes(FA2-5)',
  args: {
    ...defaultArgs,
    id: elementId,
    className: 'dummyClassname',
    lang: 'nb',
    'data-testid': '123ID',
  },
  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const container = canvas.getByRole('button');
    await expect(container).toHaveClass('dummyClassname');
    await expect(container).toHaveAttribute('id', elementId);
    await expect(container).toHaveAttribute('lang', 'nb');
    await expect(container).toHaveAttribute('data-testid', '123ID');
  },
} satisfies Story;

export const Defaults = {
  name: 'Defaults (A2, A4,A5 delvis, A7, B1, B4)',
  args: {
    ...defaultArgs,
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const container = canvas.getByRole('button');
    await expect(container).toHaveAttribute('aria-pressed', 'false');
  },
} satisfies Story;

export const WithoutCheckmark = {
  name: 'Without Checkmark (A4)',
  args: {
    showCheckmark: false,
    ...defaultArgs,
  },
  argTypes: {
    showCheckmark: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const icon = canvas.queryByRole('img');
    await expect(icon).not.toBeInTheDocument();
  },
} satisfies Story;

export const WithSelected = {
  name: 'With Selected (A6, B4)',
  args: {
    isSelected: true,
    ...defaultArgs,
  },
  argTypes: {
    showCheckmark: { table: { disable: false } },
    isSelected: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const container = canvas.getByRole('button');
    await expect(container).toHaveAttribute('aria-pressed', 'true');
  },
} satisfies Story;

export const WithSizeSmall = {
  name: 'With Size Small (A1)',
  args: {
    size: 'small',
    ...defaultArgs,
  },
  argTypes: {
    size: { table: { disable: false } },
    showCheckmark: { table: { disable: false } },
    isSelected: { table: { disable: false } },
  },
} satisfies Story;
