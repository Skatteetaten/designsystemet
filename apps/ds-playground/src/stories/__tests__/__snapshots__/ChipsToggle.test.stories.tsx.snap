// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/Chips/Toggle Defaults play-test 1`] = `
<div data-test-block="true">
  <button type="button"
          class="_toggleChip_1ard9_1  _toggleChip_withCheckmark_1ard9_43"
          aria-pressed="false"
  >
    <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_1ard9_58"
         viewbox="0 0 24 24"
         focusable="false"
         aria-hidden="true"
         role="img"
    >
      <path d="M12 20a8 8 0 1 1 0-16 8 8 0 0 1 0 16Zm0-18a10 10 0 1 0 0 20 10 10 0 0 0 0-20Z">
      </path>
    </svg>
    Trondheim
  </button>
</div>
`;

exports[`Tester/Chips/Toggle WithAttributes play-test 1`] = `
<div data-test-block="true">
  <button type="button"
          id="htmlId"
          class="_toggleChip_1ard9_1  _toggleChip_withCheckmark_1ard9_43 dummyClassname"
          lang="nb"
          data-testid="123ID"
          aria-pressed="false"
  >
    <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_1ard9_58"
         viewbox="0 0 24 24"
         focusable="false"
         aria-hidden="true"
         role="img"
    >
      <path d="M12 20a8 8 0 1 1 0-16 8 8 0 0 1 0 16Zm0-18a10 10 0 1 0 0 20 10 10 0 0 0 0-20Z">
      </path>
    </svg>
    Trondheim
  </button>
</div>
`;

exports[`Tester/Chips/Toggle WithRef play-test 1`] = `
<div data-test-block="true">
  <button type="button"
          class="_toggleChip_1ard9_1  _toggleChip_withCheckmark_1ard9_43"
          aria-pressed="false"
          id="dummyIdForwardedFromRef"
  >
    <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_1ard9_58"
         viewbox="0 0 24 24"
         focusable="false"
         aria-hidden="true"
         role="img"
    >
      <path d="M12 20a8 8 0 1 1 0-16 8 8 0 0 1 0 16Zm0-18a10 10 0 1 0 0 20 10 10 0 0 0 0-20Z">
      </path>
    </svg>
    Trondheim
  </button>
</div>
`;

exports[`Tester/Chips/Toggle WithSelected play-test 1`] = `
<div data-test-block="true">
  <button type="button"
          class="_toggleChip_1ard9_1  _toggleChip_withCheckmark_1ard9_43"
          aria-pressed="true"
  >
    <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_1ard9_58"
         viewbox="0 0 24 24"
         focusable="false"
         aria-hidden="true"
         role="img"
    >
      <path d="M12 2a10 10 0 0 1 10 10c0 5.523-4.477 10-10 10a10 10 0 0 1 0-20Zm-1 14.5 7-7-1.41-1.41L11 13.67l-3.09-3.08L6.5 12l4.5 4.5Z">
      </path>
    </svg>
    Trondheim
  </button>
</div>
`;

exports[`Tester/Chips/Toggle WithSizeSmall smoke-test 1`] = `
<div data-test-block="true">
  <button type="button"
          class="_toggleChip_1ard9_1 _toggleChip_small_1ard9_37 _toggleChip_withCheckmark_1ard9_43"
          aria-pressed="false"
  >
    <svg class="_systemIcon_6zbcy_2 _systemIcon_small_6zbcy_5 _icon_1ard9_58"
         viewbox="0 0 24 24"
         focusable="false"
         aria-hidden="true"
         role="img"
    >
      <path d="M12 20a8 8 0 1 1 0-16 8 8 0 0 1 0 16Zm0-18a10 10 0 1 0 0 20 10 10 0 0 0 0-20Z">
      </path>
    </svg>
    Trondheim
  </button>
</div>
`;

exports[`Tester/Chips/Toggle WithoutCheckmark play-test 1`] = `
<div data-test-block="true">
  <button type="button"
          class="_toggleChip_1ard9_1"
          aria-pressed="false"
  >
    Trondheim
  </button>
</div>
`;
